File: 240409/짝수만 2로 나누기/divide-even-numbers-by-2.py
N = int(input())
n_lst = list(map(int, input().split()))

def modify(lst):
    for n_idx, n in enumerate(lst):
        if n % 2 == 0:
            lst[n_idx] = n // 2
    
    return lst

print(*modify(n_lst))

File: 251220/짝수만 2로 나누기/divide-even-numbers-by-2.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<int> modify(vector<int> lst) {
    for (int i = 0; i < lst.size(); i++) {
        if (lst[i] % 2 == 0) {
            lst[i] /= 2;
        }
    }
    return lst;
}

int main() {
    int N;
    cin >> N;

    vector<int> n_lst(N);
    for (int i = 0; i < N; i++) {
        cin >> n_lst[i];
    }

    vector<int> result = modify(n_lst);

    for (int x : result) {
        cout << x << " ";
    }

    return 0;
}
